Remove square and power-of-two default flags of PVRTexTool

osman-turan opened the pull request:
Textures which are compressed by PVRTexTool are forced to square and power-of-two. Unfortunately, there is no way to override this behaviour. This fix removes default flags. So, if a user wants to use square and power-of-two textures, he can use `--flags`.

TimvanScherpenzeel commented:
Hi Osman,

Thank you for your pull request, the original use case for this tool was WebGL 1.0 which requires power-of-two textures but I've noticed that users have been using it for other use cases as well.

Unfortunately this would require a major release as it could potentially break existing workflows but I think you are right that it should not assume square power-of-two textures by default without an easy way to turn it off.

Kind regards,

Tim

osman-turan commented:
Hi Tim,

Considering your reasons, it makes sense not to incorporate these changes. Then what about this: I'll try to keep old default flags, but at the same time I'll provide a way to override if needed. What do you think?

Osman

osman-turan committed:
Add square and pot flags to control texture resizing

Textures which are compressed by PVRTexTool are forced to square and power-of-two. Unfortunately, there is no way to override this behaviour. This commit introduces 2 new optional flags to address this issue: `--square` and `--pot`. Both flags accept a parameter to specify whether the texture should be resized smaller (`-`), larger (`+`) or no change at all (`no`). Default parameter is `+` for both flags due to backward compatibility.

osman-turan commented:
@TimvanScherpenzeel I've updated the code to keep backward compatibility by introducing 2 new optional flags. These flags can also control how resizing should be done: downsize or upsize.

TimvanScherpenzeel commented:
Hi @osman-turan, thank you for your update. My apologies for the late response, I've been very busy with work related projects lately.

TimvanScherpenzeel commented:
I've just released version `texture-compressor@1.0.1` on NPM: https://www.npmjs.com/package/texture-compressor

osman-turan commented:
Thank you for the approval and fast NPM deployment! Now, it's time to update my NPM dependencies 🙂 Thank you again for this wonderful project.
